### PR TITLE
test: unit test of class Memory in Module_base

### PR DESCRIPTION
**Pull request by hongriTianqi**

### CODE REVIEW

File: source/module_base/test/memory_test.cpp [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+	// for output in file
+	std::ofstream ofs;
+	std::ifstream ifs;
+	void TearDown()
+	{
+		remove("tmp");
+	}
+};
+
+TEST_F(MemoryTest, calculatemem)
```
Comment (caic99) [inline source/module_base/test/memory_test.cpp:46]: `CalculateMem` would be better.

File: source/module_base/memory.cpp [hunk 2]
```diff
@@ -6,17 +6,17 @@
...
 // 1024 KB   = 1 MB
 // 1024 MB   = 1 GB
 double Memory::total = 0.0;
-double Memory::complex_matrix_memory = 16.0;//(Byte)
-double Memory::double_memory = 8.0;//(Byte)
-double Memory::int_memory = 4.0;//(Byte)
-double Memory::bool_memory = 1.0;//(Byte)
-double Memory::float_memory = 4.0;//(Byte)
-double Memory::short_memory = 2.0;//(Byte)
+double Memory::complex_matrix_memory = 2*sizeof(double); //16.0 (Byte)
```
Comment (caic99) [inline source/module_base/memory.cpp:14]: Why these memory definitions are in `double`?

File: source/module_base/memory.cpp [hunk 3]
```diff
@@ -6,17 +6,17 @@
...
+double Memory::double_memory = sizeof(double); //8.0 (Byte)
+double Memory::int_memory = sizeof(int); //4.0 (Byte)
+double Memory::bool_memory = sizeof(bool); // 1.0 (Byte)
+double Memory::float_memory = sizeof(float);// 4(Byte)
+double Memory::short_memory = sizeof(short); //2.0 (Byte)
 
 int Memory::n_memory = 500;
```
Comment (caic99) [inline source/module_base/memory.cpp:21]: What does this variable mean?

File: source/module_base/memory.cpp (hunk 2 above)
Comment (hongriTianqi) [reply source/module_base/memory.cpp:14]: double for easier afterwards multiplication and addition. For example, when we calculate the memory requirements of a large matrix, we need to multiply the size of the matrix with the memory requirement for each element.

File: source/module_base/memory.cpp (hunk 3 above)
Comment (hongriTianqi) [reply source/module_base/memory.cpp:21]: maximum number of (class+calculation type) composites that need memory records. I think it is only a temporary expectation of the maximum number, but may need to be increased in future, or be set in a better way. Not added by me.

File: source/module_base/test/memory_test.cpp (hunk 1 above)
Comment (hongriTianqi) [reply source/module_base/test/memory_test.cpp:46]: OK, agree. I will change it in due course.

**Comment by caic99** [comment]
This PR is re-submitted and merged in #680 .

**Pull request closed without merging**